sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1312,6 +1311,40 @@ void ProgramManager::addImages(pi_device_binaries DeviceBinary) {
           }
         }
       }
+      // ... and initialize associated host_pipe information
+      {
+        std::lock_guard HostPipesGuard(m_HostPipesMutex);
+        auto HostPipes = Img->getHostPipes();
+        for (const pi_device_binary_property &HostPipe : HostPipes) {
+          ByteArray HostPipeInfo = DeviceBinaryProperty(HostPipe).asByteArray();
+
+          // The supplied host_pipe info property is expected to contain:
+          // * 8 bytes - Size of the property.
+          // * 4 bytes - Size of the underlying type in the host_pipe.
+          // Note: Property may be padded.
+          constexpr unsigned int NumPropertySizeBytes = 8;
+          constexpr unsigned int NumTypeBytes = 4;
+          assert(HostPipeInfo.size() >= NumPropertySizeBytes + NumTypeBytes &&
+                 "Unexpected property size");
+          auto TypeSize = *reinterpret_cast<const std::uint32_t *>(
+              &HostPipeInfo[NumPropertySizeBytes]);
```

Comment:
I would be better to use the `dropBytes` and `consume` member functions on `ByteArray` here to avoid strict aliasing violations. You can see these being used a couple lines further up for `DeviceGlobalInfo`.